src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php

```diff
@@ -37,6 +37,36 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
     /**
      * {@inheritDoc}
      *
+     * The ORM implementation for preparing the object
+     */
+    public function prepareObject(TypeInterface $type, $parent = null)
+    {
+        $object = parent::prepareObject($type);
+
+        if (null == $parent){
+            throw new RuntimeException('You need a parent to create new objects');
+        }
+
+        /** @var \Doctrine\ORM\Mapping\ClassMetadata $metaData */
+        $metaData = $this->om->getClassMetaData(get_class($object));
+        $metaDataParent = $this->om->getClassMetaData(get_class($parent));
+
+        foreach ($metaData->associationMappings as $mapping)
```

Comment:
the use case i think of is that e.g. my class has references to objects of the same class once as parent and once as children. then i would need to have custom logic as the generic logic fails. 

i would pass $object and $metadata. we are in an orm context, so it does not hurt to have it.